internal/mailer/mailme.go

```diff
@@ -82,8 +84,45 @@ func (m *MailmeMailer) Mail(to, subjectTemplate, templateURL, defaultTemplate st
 	if m.LocalName != "" {
 		dial.LocalName = m.LocalName
 	}
-	return dial.DialAndSend(mail)
 
+	defer func() {
+		projectRef, ok := getProjectRefFromHeaders(headers)
+		if !ok {
+			return
+		}
+
+		fields := logrus.Fields{
+			"event":         "mail.send",
+			"project_ref":   projectRef,
+			"mail_from":     m.From,
+			"mail_to":       to,
+			"mail_subject":  subject,
+			"mail_template": subjectTemplate,
+		}
+		m.Logger.WithFields(fields).Info("mail.send")
+	}()
+	if err := dial.DialAndSend(mail); err != nil {
+		return err
+	}
+	return nil
+}
+
+const (
+	mailSupabaseProjectRefHeader = `x-supabase-project-ref`
+)
+
+func getProjectRefFromHeaders(headers map[string][]string) (string, bool) {
+	for key, val := range headers {
+		if len(val) == 0 || val[0] == "" {
+			continue
+		}
+
+		hdr := strings.ToLower(key)
+		if hdr == mailSupabaseProjectRefHeader {
+			return val[0], true
+		}
+	}
+	return "", false
```

Comment:
likewise, i don't think we should leak the concept of a "project ref" to the auth service